File: Dart/Framework/metrics/space/memory_usage.dart
class MemoryUsage {
  int getTotalMemory() {
    return _getRuntime().totalMemory;
  }

  int getFreeMemory() {
    return _getRuntime().freeMemory;
  }

  int getMemoryUsage() {
    return _getRuntime().totalMemory - _getRuntime().freeMemory;
  }

  Runtime _getRuntime() {
    return Runtime();
  }
}

class Runtime {
  final int totalMemory;
  final int freeMemory;

  Runtime({this.totalMemory = 0, this.freeMemory = 0});
}

File: Old_Experiments/Java/Framework/src/metrics/space/MemoryUsage.java
package metrics.space;

public class MemoryUsage {



    //MainMemory
    //AuxiliaryMemory
    //Cache


        public long getTotalMemory(){
            return Runtime.getRuntime().totalMemory();
        }
        public long getFreeMemory(){
            return Runtime.getRuntime().freeMemory();
        }
        public long getMemoryUsage(){
            return Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
        }
}


